libs/shared-web/src/providers/LaunchDarklyProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import { IOrganizationEntity } from '@novu/shared';
+import { asyncWithLDProvider } from 'launchdarkly-react-client-sdk';
+import { PropsWithChildren, ReactNode, useEffect, useRef, useState } from 'react';
+import { LAUNCH_DARKLY_CLIENT_SIDE_ID } from '../config';
+import { useFeatureFlags } from '../hooks';
+import { checkShouldUseLaunchDarkly } from '../utils';
+import { useAuthContext } from './AuthProvider';
+
+/** A provider with children required */
+type GenericLDProvider = Awaited<ReturnType<typeof asyncWithLDProvider>>;
+
+/** Simply renders the children */
+const DEFAULT_GENERIC_PROVIDER: GenericLDProvider = ({ children }) => <>{children}</>;
+
+export interface ILaunchDarklyProviderProps {
+  /** Renders when LaunchDarkly is enabled and is awaiting initialization */
+  fallbackDisplay: ReactNode;
+}
+
+/**
+ * Async provider for feature flags.
+ *
+ * @requires AuthProvider must be wrapped in the AuthProvider.
+ */
+export const LaunchDarklyProvider: React.FC<PropsWithChildren<ILaunchDarklyProviderProps>> = ({
+  children,
+  fallbackDisplay,
+}) => {
+  const LDProvider = useRef<GenericLDProvider>(DEFAULT_GENERIC_PROVIDER);
+  const [isLDReady, setIsLDReady] = useState<boolean>(false);
```

Comment:
🤓 **nitpick (non-blocking):** the same could be done with just `useState` ;)

Reply:
My thinking was to avoid passing around the provider when a boolean would suffice -- especially now that we're using some "selectors"